airbyte-lib/airbyte_lib/datasets/_sql.py

```diff
@@ -85,9 +93,11 @@ class CachedDataset(SQLDataset):
     """
 
     def __init__(self, cache: SQLCacheBase, stream_name: str) -> None:
-        self._cache: SQLCacheBase = cache
-        self._stream_name: str = stream_name
-        self._query_statement: Selectable = self.to_sql_table().select()
+        super().__init__(
+            cache=cache,
+            stream_name=stream_name,
+            query_statement=self.to_sql_table().select(),
```

Comment:
This doesn't work, because `to_sql_table` requires `_cache` to be set.

Reply:
Good catch! Thanks!

Reply:
Fixed now.